splunklogger.js

```diff
@@ -474,27 +655,32 @@ SplunkLogger.prototype.send = function (context, callback) {
  * @param {function} [callback] - A callback function: <code>function(err, response, body)</code>.
  * @public
  */
-SplunkLogger.prototype.flush = function (callback) {
+SplunkLogger.prototype.flush = function(callback) {
     callback = callback || function(){};
 
     var context = {};
 
-    // Use the batching setting from this.config
-    if (this.config.autoFlush) {
-        // TODO: handle case of multiple events with autoFlush off, flushing fast
-        // Just take the oldest event in the queue
-        context = this.contextQueue.pop();
-    }
-    else {
-        // Empty the event queue
+    var batchOverSize = this.config.maxBatchSize > 0 && this.calculateBatchSize() > this.config.maxBatchSize;
+    var batchOverCount = this.config.maxBatchCount > 0 && this.contextQueue.length >= this.config.maxBatchCount;
+    var isBatched = !this.config.autoFlush || (this.config.autoFlush && (batchOverSize || batchOverCount || this._timerID));
+
+    // Empty the queue if manual/interval/size/count batching
+    if (isBatched) {
         var queue = this.contextQueue;
         this.contextQueue = [];
+        this.eventSizes = [];
         var data = "";
         for (var i = 0; i < queue.length; i++) {
             data += JSON.stringify(this._makeBody(queue[i]));
         }
         context.message = data;
     }
+    // Just take the oldest event in the queue
+    else {
+        // TODO: handle case of multiple events with autoFlush off, flushing fast
```

Comment:
I don't understand this case. If I have autoFlush off, and I manually call `flush()`, it should just send everything thats queued up at the moment.